File: src/components/exercise-images.ts
// Photos d'illustration des mouvements de musculation.
//
// Chaque exercice est illustré par une paire de photos : position de DÉPART
// puis position FINALE du mouvement, ce qui montre précisément l'amplitude à
// réaliser. Les images viennent de free-exercise-db (domaine public, Unlicense)
// et sont **bundlées dans l'app** : aucune dépendance réseau, conforme à
// l'approche local-first du projet.
//
// `require()` doit recevoir un chemin statique (contrainte du bundler Metro),
// d'où ce registre clé → paire d'images plutôt qu'un chemin construit à la volée.

// Les assets bundlés via `require()` se résolvent en identifiants numériques
// de module : on type donc la paire en `number`, accepté par <Image> d'expo-image.

/** Paire [départ, fin] de photos illustrant un mouvement. */
export type ExerciseImagePair = readonly [number, number];

const IMAGES: Record<string, ExerciseImagePair> = {
  'goblet-squat': [
    require('../../assets/images/exercises/goblet-squat-0.jpg'),
    require('../../assets/images/exercises/goblet-squat-1.jpg'),
  ],
  'dumbbell-bench-press': [
    require('../../assets/images/exercises/dumbbell-bench-press-0.jpg'),
    require('../../assets/images/exercises/dumbbell-bench-press-1.jpg'),
  ],
  'one-arm-dumbbell-row': [
    require('../../assets/images/exercises/one-arm-dumbbell-row-0.jpg'),
    require('../../assets/images/exercises/one-arm-dumbbell-row-1.jpg'),
  ],
  'dumbbell-lunges': [
    require('../../assets/images/exercises/dumbbell-lunges-0.jpg'),
    require('../../assets/images/exercises/dumbbell-lunges-1.jpg'),
  ],
  plank: [
    require('../../assets/images/exercises/plank-0.jpg'),
    require('../../assets/images/exercises/plank-1.jpg'),
  ],
  'dumbbell-romanian-deadlift': [
    require('../../assets/images/exercises/dumbbell-romanian-deadlift-0.jpg'),
    require('../../assets/images/exercises/dumbbell-romanian-deadlift-1.jpg'),
  ],
  'standing-dumbbell-press': [
    require('../../assets/images/exercises/standing-dumbbell-press-0.jpg'),
    require('../../assets/images/exercises/standing-dumbbell-press-1.jpg'),
  ],
  'bent-over-two-dumbbell-row': [
    require('../../assets/images/exercises/bent-over-two-dumbbell-row-0.jpg'),
    require('../../assets/images/exercises/bent-over-two-dumbbell-row-1.jpg'),
  ],
  'bulgarian-split-squat': [
    require('../../assets/images/exercises/bulgarian-split-squat-0.jpg'),
    require('../../assets/images/exercises/bulgarian-split-squat-1.jpg'),
  ],
  'side-plank': [
    require('../../assets/images/exercises/side-plank-0.jpg'),
    require('../../assets/images/exercises/side-plank-1.jpg'),
  ],
};

/** Photos [départ, fin] pour une clé d'illustration, ou `null` si inconnue. */
export function exerciseImages(key: string | undefined): ExerciseImagePair | null {
  if (!key) return null;
  return IMAGES[key] ?? null;
}

File: app/src/main/java/ovh/battistella/elan/ui/components/ExerciseImages.kt
// Illustrations photo des exercices (paire départ → fin), résolues localement
// depuis les ressources bundlées (`res/drawable-nodpi/exercise_<clé>_{0,1}.jpg`).
// Photos domaine public (free-exercise-db) : aucun accès réseau.
package ovh.battistella.elan.ui.components

import androidx.annotation.DrawableRes
import ovh.battistella.elan.R

/** Les deux positions d'un mouvement : départ puis fin. */
data class ExercisePhotos(@DrawableRes val start: Int, @DrawableRes val end: Int)

private val EXERCISE_IMAGES: Map<String, ExercisePhotos> = mapOf(
    "goblet-squat" to ExercisePhotos(R.drawable.exercise_goblet_squat_0, R.drawable.exercise_goblet_squat_1),
    "dumbbell-bench-press" to ExercisePhotos(R.drawable.exercise_dumbbell_bench_press_0, R.drawable.exercise_dumbbell_bench_press_1),
    "one-arm-dumbbell-row" to ExercisePhotos(R.drawable.exercise_one_arm_dumbbell_row_0, R.drawable.exercise_one_arm_dumbbell_row_1),
    "dumbbell-lunges" to ExercisePhotos(R.drawable.exercise_dumbbell_lunges_0, R.drawable.exercise_dumbbell_lunges_1),
    "plank" to ExercisePhotos(R.drawable.exercise_plank_0, R.drawable.exercise_plank_1),
    "dumbbell-romanian-deadlift" to ExercisePhotos(R.drawable.exercise_dumbbell_romanian_deadlift_0, R.drawable.exercise_dumbbell_romanian_deadlift_1),
    "standing-dumbbell-press" to ExercisePhotos(R.drawable.exercise_standing_dumbbell_press_0, R.drawable.exercise_standing_dumbbell_press_1),
    "bent-over-two-dumbbell-row" to ExercisePhotos(R.drawable.exercise_bent_over_two_dumbbell_row_0, R.drawable.exercise_bent_over_two_dumbbell_row_1),
    "bulgarian-split-squat" to ExercisePhotos(R.drawable.exercise_bulgarian_split_squat_0, R.drawable.exercise_bulgarian_split_squat_1),
    "side-plank" to ExercisePhotos(R.drawable.exercise_side_plank_0, R.drawable.exercise_side_plank_1),
)

/** Paire de photos d'une clé d'illustration (`imageKey` du programme / catalogue), ou `null`. */
fun exerciseImages(imageKey: String?): ExercisePhotos? = if (imageKey == null) null else EXERCISE_IMAGES[imageKey]
